src/AbstractDoctrineAnnotationFixer.php

```diff
@@ -0,0 +1,234 @@
+     * Fixes Doctrine annotations from the given PHPDoc style comment.
+     *
+     * @param DoctrineAnnotationTokens $doctrineAnnotationTokens
+     */
+    abstract protected function fixAnnotations(DoctrineAnnotationTokens $doctrineAnnotationTokens);
+
+    /**
+     * @return string[]
+     */
+    protected function getDefaultIgnoredTags()
+    {
+        return array_merge(Tag::getTags(), array(
+            // PHPUnit
+            'after',
+            'afterClass',
+            'backupGlobals',
+            'backupStaticAttributes',
+            'before',
+            'beforeClass',
+            'codeCoverageIgnore',
+            'codeCoverageIgnoreStart',
+            'codeCoverageIgnoreEnd',
+            'covers',
+            'coversDefaultClass',
+            'coversNothing',
+            'dataProvider',
+            'depends',
+            'expectedException',
+            'expectedExceptionCode',
+            'expectedExceptionMessage',
+            'expectedExceptionMessageRegExp',
+            'group',
+            'large',
+            'medium',
+            'preserveGlobalState',
+            'requires',
+            'runTestsInSeparateProcesses',
+            'runInSeparateProcess',
+            'small',
+            'test',
+            'testdox',
+            'ticket',
+
+            // other
+            'TODO',
+            'todo',
+            'FIXME',
+            'fixme',
+        ));
+    }
+
+    /**
+     * @param PhpTokens $tokens
+     * @param int       $index
+     *
+     * @return bool
+     */
+    private function nextElementAcceptsDoctrineAnnotations(PhpTokens $tokens, $index)
+    {
+        do {
+            $index = $tokens->getNextMeaningfulToken($index);
+        } while ($tokens[$index]->isGivenKind(array(T_ABSTRACT, T_FINAL)));
+
+        if ($tokens[$index]->isClassy()) {
+            return true;
+        }
+
+        while ($tokens[$index]->isGivenKind(array(T_PUBLIC, T_PROTECTED, T_PRIVATE, T_FINAL, T_ABSTRACT))) {
+            $index = $tokens->getNextMeaningfulToken($index);
+        }
+
+        $analyzer = new TokensAnalyzer($tokens);
+
+        return array_key_exists($index, $analyzer->getClassyElements());
+    }
+
+    /**
+     * @param PhpToken $input
+     *
+     * @return DoctrineAnnotationTokens
+     */
+    private function getDoctrineAnnotationTokens(PhpToken $input)
+    {
+        $tokens = new DoctrineAnnotationTokens();
+        $tokens->setIgnoredTags($this->ignoredTags);
+
+        $lexer = new DocLexer();
+        $lexer->setInput($content = $input->getContent());
+        $lastPosition = 0;
+        while (null !== $token = $lexer->peek()) {
+            if (DocLexer::T_NONE === $token['type']) {
+                continue;
+            }
+
+            if (DocLexer::T_STRING === $token['type']) {
+                $token['value'] = '"'.str_replace('"', '""', $token['value']).'"';
+            }
+
+            if ($lastPosition !== $newPosition = $token['position']) {
```

Comment:
yes, all of them.
I just found it pointless to complain about every single them individually, making a lot of spam comments ;)